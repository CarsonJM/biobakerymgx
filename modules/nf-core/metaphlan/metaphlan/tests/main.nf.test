nextflow_process {

    name "Test Process: METAPHLAN_METAPHLAN"
    script "../main.nf"
    process "METAPHLAN_METAPHLAN"

    tag "modules"
    tag "modules_local"
    tag "metaphlan"
    tag "metaphlan/makedb"
    tag "metaphlan/metaphlan"

    setup {
        run("METAPHLAN_MAKEDB") {
            script "../../makedb/main.nf"
            process {
                """
                input[0] = "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103"
                """
            }
        }
    }


    test("fastq.gz + metaphlan_db") {

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz", checkIfExists: true),
                            file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz", checkIfExists: true)
                        ]
                    ]
                )
                input[1] = METAPHLAN_MAKEDB.out.db
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    process.out.profile,
                    process.out.bt2out,
                    process.out.versions
                ).match() },
                { assert process.out.biom != null }
            )
        }
    }

    test("fastq.gz + metaphlan_db -stub ") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz", checkIfExists: true),
                            file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz", checkIfExists: true)
                        ]
                    ]
                )
                input[1] = METAPHLAN_MAKEDB.out.db
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
