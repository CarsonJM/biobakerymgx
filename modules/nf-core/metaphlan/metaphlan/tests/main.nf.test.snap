{
    "fastq.gz + metaphlan_db -stub ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_profile.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.biom:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bowtie2out.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,2a9ed777cd0065f872877f20553df297"
                ],
                "biom": [
                    [
                        {
                            "id": "test"
                        },
                        "test.biom:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bt2out": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bowtie2out.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "profile": [
                    [
                        {
                            "id": "test"
                        },
                        "test_profile.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2a9ed777cd0065f872877f20553df297"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T10:03:39.92337057"
    },
    "fastq.gz + metaphlan_db": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_profile.txt:md5,cd4748da44ffcd985244ffd6103f6a6d"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test.bowtie2out.txt:md5,9bf563c43769e8300be64a8bf1a03c71"
                ]
            ],
            [
                "versions.yml:md5,2a9ed777cd0065f872877f20553df297"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T10:03:21.004630639"
    }
}